### ISSUE TITLE
[HttpClient] cURL timeout requests block other requests

### ISSUE THREAD

**Issue by Toflar** [symfony/symfony#34933]
**Symfony version(s) affected**: 4.4.*

**Description**  

I don't know if this is actually fixable or not but if a requests runs into a timeout, non of the other requests streams any data until that request reaches the timeout.

**How to reproduce**  

```php
<?php

include_once 'vendor/autoload.php';

$client = new \Symfony\Component\HttpClient\CurlHttpClient(['max_duration' => 5]);

$urls = ['http://httpstat.us/200?sleep=6000', 'http://httpstat.us/200'];
$responses = [];

foreach ($urls as $url) {
    $responses[] = $client->request('GET', $url, ['user_data' => $url]);
}

foreach($client->stream($responses) as $response => $chunk) {
    var_dump($response->getInfo('user_data'));
}
```

As you can see, I configured the first request to sleep 6 seconds which means it runs into a timeout. But until that happens, I would expect to get the chunks of the second request. However, I don't get any of those.

**Possible Solution**  

Not sure if even fixable. Requires better knowledge of the underlying cURL implementation, I guess.

**Comment by nicolas-grekas** [comment]
Works for me. If I `dump($chunk)` I see this after 5s:

```
^ Symfony\Component\HttpClient\Chunk\FirstChunk^ {#18
  -offset: 0
  -content: ""
}
^ Symfony\Component\HttpClient\Chunk\DataChunk^ {#14
  -offset: 6
  -content: "200 OK"
}
^ Symfony\Component\HttpClient\Chunk\LastChunk^ {#13
  -offset: 6
  -content: ""
}
^ Symfony\Component\HttpClient\Chunk\ErrorChunk^ {#8
  -didThrow: false
  -offset: 0
  -errorMessage: "Timeout was reached for "http://httpstat.us/200?sleep=6000"."
  -error: Symfony\Component\HttpClient\Exception\TransportException^ {#12
    #message: "Timeout was reached for "http://httpstat.us/200?sleep=6000"."
    #code: 0
    #file: "./src/Symfony/Component/HttpClient/Response/CurlResponse.php"
    #line: 320
    trace: {
      ./src/Symfony/Component/HttpClient/Response/CurlResponse.php:320 {
        Symfony\Component\HttpClient\Response\CurlResponse::perform(CurlClientState $multi, array &$responses = null): void^
        ›     $multi->handlesActivity[$id][] = null;
        ›     $multi->handlesActivity[$id][] = \in_array($result, [\CURLE_OK, \CURLE_TOO_MANY_REDIRECTS], true) || '_0' === $waitFor || curl_getinfo($ch, CURLINFO_SIZE_DOWNLOAD) === curl_getinfo($ch, CURLINFO_CONTENT_LENGTH_DOWNLOAD) ? null : new TransportException(sprintf('%s for "%s".', curl_strerror($result), curl_getinfo($ch, CURLINFO_EFFECTIVE_URL)));
        › }
      }
      ./src/Symfony/Component/HttpClient/Response/ResponseTrait.php:305 { …}
      Symfony\Component\HttpClient\Response\CurlResponse::stream() {}
      ./src/Symfony/Component/HttpClient/Response/ResponseStream.php:42 { …}
      ./test.php:14 { …}
    }
  }
}
```

**Comment by Toflar** [comment]
What the heck? I increased the limits and it worked. So I checked the `total_time` and noticed that these requests all took at least 5 seconds to complete. I googled a bit and found that there are actually quite a few issues/problems, apparently due to DNS. Even the number `5` turns up quite often. Funnily enough, this doesn't happen in the browser so I wonder if there's some specific cURL filtering somewhere but I can't really imagine. If I configure my DNS settings to `1.1.1.1` it works just fine.
So all good on Symfony side here. I'm closing the issue then, it might still serve as info for other devs having similar issues :) Thanks for trying to reproduce!
